File: scripts/fixtures/comment-hygiene/seeded.cpp
// Fixture source for scripts/check-comment-hygiene.mjs. Not compiled — this
// directory is outside every build in the repository.
//
// Two seeded findings, one per class, in the dialect the foobar shim is written
// in. The shim is compiled separately from the core and drifts the same way, so
// the gate reads it through the same two classes; what differs is the lexer, and
// the silences below are the three places C and Rust disagree. Enumerated in
// ../README.md.

// SEEDED, class 1 — a relative link in a C++ comment. Broken on purpose, since
// a link that resolves rots exactly the same way.
//
// [the decision]: ../../../docs/adrs/0127-not-a-real-path.md

// SEEDED, class 2 — plan-relative narration, in the spelling the shim reaches
// for: `visible` used to be a latch driven by window messages alone.
static const float kSeededSpan = 1.0f;

/* SILENT — a C block comment does NOT nest. The /* below is an ordinary two
   characters inside a comment, and the first */
/* close ends it. A lexer running Rust's nesting rules here would swallow the
   rest of the file and report nothing at all, which is the failure mode this
   case exists to catch. */
static const float kSilentBlock = 2.0f;

// SILENT — a bare-number citation, which is the form ADR-0127 replaces the
// links WITH. Plan 0110 added the roster call and ADR-0117 argued its shape.
static const int kSilentCitation = 3;

// SILENT — a `//` inside a string is not a comment, in both of C++'s spellings.
// The raw form takes a caller-chosen delimiter rather than Rust's hash count,
// which is the second place the two lexers diverge.
static const char* kSilentStrings[] = {
    "https://example.invalid/../not/a/link",
    R"tag(// the plan used to live here)tag",
    "a \" quote that opens nothing // and this plan is not narration",
};

// SILENT — a char literal. C has no lifetimes, so every `'` opens one; Rust's
// `'a` does not, and that is the third divergence.
static const char kSilentChar = '"';
static const char kSilentEscape = '\'';

// SILENT — an escaped false positive, same escape and same two-line reach as
// the Rust side.
//
// hygiene-allow: quoting the vocabulary the gate rejects
// The handle used to be freed here and is no longer.
static const float kSilentEscaped = 4.0f;

File: scripts/fixtures/comment-hygiene/seeded.rs
// Fixture source for scripts/check-comment-hygiene.mjs. Not compiled — this
// directory is outside every crate in the workspace.
//
// Two seeded findings, one per class, and four silences that matter more than
// the findings do. The silences are enumerated in ../README.md.

// SEEDED, class 1 — a relative link in a Rust comment. Broken on purpose, since
// a link that resolves rots exactly the same way.
//
// [the decision]: ../../../docs/adrs/0127-not-a-real-path.md

// SILENT — a rustdoc intra-doc link. rustc resolves this, so it cannot rot
// silently, and it is the linking mechanism ADR-0127 keeps: [Seeded::render],
// and the labelled form [the renderer](crate::render::Renderer).

// SILENT — a bare-number citation, which is the form ADR-0127 replaces the links
// WITH. Plan 0045 Phase 3 argued the knee, and the plans README rosters it. A
// gate that fired on the word "plan" would convict its own fix.
//
// SILENT for the same reason, in the possessive spelling a sentence
// reaches for when the citation is the subject: the Plan 0045 Phase 4b
// defect is reachable from a bound expression.
pub const SEEDED_KNEE: f32 = 0.6;

/// SEEDED, class 2 — plan-relative narration. Once a session closes there is no
/// "this plan" to resolve against; there is only the code. Exactly one finding
/// on this line: every other rejected phrase is seeded in a file of its own, so
/// a count that moves names which form moved.
pub const SEEDED_SPAN: f32 = 1.0;

// SILENT — an escaped false positive. The word list has them, and the escape
// covers its own line and the one after it.
//
// hygiene-allow: quoting the vocabulary the gate rejects
// The tonemap value used to compute the knee is no longer read here.
pub const SEEDED_ESCAPED: f32 = 2.0;

// SILENT — a `//` inside a string is not a comment, and neither is a `"` inside
// one. This is why the checker lexes Rust rather than grepping lines.
pub fn seeded_strings() -> [&'static str; 3] {
    [
        "https://example.invalid/../not/a/link",
        r#"// the plan used to live here"#,
        "a \" quote that opens nothing // and this plan is not narration",
    ]
}
